src/importlinter/contracts/layers.py

```diff
@@ -392,7 +392,7 @@ def _pop_direct_imports(cls, higher_layer_package, lower_layer_package, graph: I
                             {
                                 "importer": lower_layer_module,
                                 "imported": imported_module,
-                                "line_number": "?",
+                                "line_number": -1,
```

Comment:
I've just pushed an update to adjust the report rendering code, and to remove a couple more unnecessary `type: ignore` comments.

Reply:
Nicely done!

I'd feel more comfortable merging if this had test coverage of this particular case. Is this something you could look into? Otherwise I should be able to get around to it. It should be possible to trigger by checking and rendering a contract on a graph that has some illegal imports added manually to the graph without import details.

Reply:
I'm not sure I'd be able to effectively do that sorry. Please feel free to push additional changes to my branch.

Reply:
No problem, will do.

To manage expectations, I might not be able to get onto this for a couple of weeks. Feel free to chase if you haven't seen anything in 3 weeks 😄